dynparquet/hashed.go

```diff
@@ -105,19 +105,59 @@ func HashArray(arr arrow.Array) []uint64 {
 func hashListArray(arr *array.List) []uint64 {
 	res := make([]uint64, arr.Len())
 	digest := xxhash.New()
-
-	dictionaryList, binaryDictionaryList, err := arrowutils.ToConcreteList(arr)
```

Comment:
Wondering if the removal of this has performance implications? I recall it being put in to reduce the number of allocations that were performed

Reply:
> Wondering if the removal of this has performance implications?

I don't think so, we really didn't remove it. We just extended it to cover other cases. We still support dictionaries of `*array.Binary`, we just don't mendate it.

- For list of binary dictionaries nothing has changed (before this commit and now)
- For allocations , in  `int64` ,`float64` and `bool` we reuses buffers on the stack so allocation wise they are okay.

Reply:
So I ran benchmarks. there is a regression. I am fixing it now.